vcd/resource_vcd_org_user.go

```diff
@@ -0,0 +1,352 @@
+				Optional: true,
+				ForceNew: false,
+				Description: "True if the user account has been locked due to too many invalid login attempts. " +
+					"A locked user account can be re-enabled by updating the user with this flag set to false. " +
+					"Only the system can set the value to true.",
+			},
+			"take_ownership": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				ForceNew:    false,
+				Description: "Take ownership of user's objects on deletion.",
+			},
+			"deployed_vm_quota": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     10,
+				ForceNew:    false,
+				Description: "Quota of vApps that this user can deploy. A value of 0 specifies an unlimited quota.",
+			},
+			"stored_vm_quota": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     10,
+				ForceNew:    false,
+				Description: "Quota of vApps that this user can store. A value of 0 specifies an unlimited quota.",
+			},
+		},
+	}
+}
+
+func resourceToUserData(d *schema.ResourceData, meta interface{}) (*govcd.OrgUserConfiguration, *govcd.AdminOrg, error) {
+	vcdClient := meta.(*VCDClient)
+	orgName := d.Get("org").(string)
+	if orgName == "" {
+		orgName = vcdClient.Org
+	}
+	if orgName == "" {
+		return nil, nil, fmt.Errorf("missing org name")
+	}
+	adminOrg, err := govcd.GetAdminOrgByName(vcdClient.VCDClient, orgName)
+	if err != nil {
+		return nil, nil, err
+	}
+	if adminOrg.AdminOrg == nil || adminOrg.AdminOrg.HREF == "" {
+		return nil, nil, fmt.Errorf("error retrieving org %s", orgName)
+	}
+
+	var userData govcd.OrgUserConfiguration
+	userData.RoleName = d.Get("role").(string)
+
+	userData.Name = d.Get("name").(string)
+	userData.Description = d.Get("description").(string)
+	userData.FullName = d.Get("full_name").(string)
+	userData.EmailAddress = d.Get("email_address").(string)
+	userData.Telephone = d.Get("telephone").(string)
+	userData.ProviderType = d.Get("provider_type").(string)
+	userData.IsEnabled = d.Get("is_enabled").(bool)
+	userData.IsLocked = d.Get("is_locked").(bool)
+	userData.DeployedVmQuota = d.Get("deployed_vm_quota").(int)
+	userData.StoredVmQuota = d.Get("stored_vm_quota").(int)
+	userData.IM = d.Get("instant_messaging").(string)
+
+	password := d.Get("password").(string)
+	if password != "" {
+		userData.Password = password
+	}
+	passwordFile := d.Get("password_file").(string)
+	if passwordFile != "" {
+		passwordBytes, err := ioutil.ReadFile(passwordFile)
+		if err != nil {
+			return nil, nil, err
+		}
+		passwordStr := strings.TrimSpace(string(passwordBytes))
+		if passwordStr != "" {
+			userData.Password = passwordStr
+		}
+	}
+
+	if userData.Password == "" {
+		return nil, nil, fmt.Errorf(`no password provided in either "password"" or "password_file"`)
+	}
+	return &userData, &adminOrg, nil
+}
+
+func resourceToOrgUser(d *schema.ResourceData, meta interface{}) (*govcd.OrgUser, *govcd.AdminOrg, error) {
```

Comment:
Short comment above the function's purpose would be nice.

Reply:
done